prov/efa/src/rdm/rxr_srx.c

```diff
@@ -0,0 +1,188 @@
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "efa.h"
+#include "rxr_srx.h"
+
+/**
+ * @brief This call is invoked by the peer provider to obtain the receive
+ * buffers where an incoming message should be placed.
+ *
+ * @param[in] srx the fid_peer_srx
+ * @param[in] addr the source address of the incoming message
+ * @param[in] size the size of the incoming message (ignored unless multi-receive)
+ * @param[out] rx_entry the obtained receive buffer
+ * @return int 0 on success, a negative integer on failure
+ */
+static int rxr_srx_get_msg(struct fid_peer_srx *srx, fi_addr_t addr,
+		       size_t size, struct fi_peer_rx_entry **rx_entry)
+{
+    return -FI_ENOSYS;
+}
+
+/**
+ * @brief This call is invoked by the peer provider to obtain the receive
+ * buffers where an incoming tagged message should be placed.
+ *
+ * @param[in] srx the fid_peer_srx
+ * @param[in] addr the source address of the incoming message
+ * @param[in] size the size of the incoming message (ignored unless multi-receive)
+ * @param[out] rx_entry the obtained receive buffer
+ * @return int 0 on success, a negative integer on failure
+ */
+static int rxr_srx_get_tag(struct fid_peer_srx *srx, fi_addr_t addr,
+			uint64_t tag, struct fi_peer_rx_entry **rx_entry)
+{
+    return -FI_ENOSYS;
+}
+
+/**
+ * @brief This call is invoked by the peer provider to queue
+ * a received message to owner provider's unexpected queue,
+ * when the peer calls owners->get_msg() but the owner fails to
+ * find a matching receive buffer.
+ *
+ * @param[in] rx_entry the entry to be queued
+ * @return int 0 on success, a negative integer on failure
+ */
+static int rxr_srx_queue_msg(struct fi_peer_rx_entry *rx_entry)
+{
+    return -FI_ENOSYS;
+}
+
+/**
+ * @brief This call is invoked by the peer provider to queue
+ * a received tagged message to owner provider's unexpected queue,
+ * when the peer calls owners->get_msg() but the owner fails to
+ * find a matching receive buffer.
+ *
+ * @param[in] rx_entry the entry to be queued
+ * @return int 0 on success, a negative integer on failure
+ */
+static int rxr_srx_queue_tag(struct fi_peer_rx_entry *rx_entry)
+{
+	return -FI_ENOSYS;
+}
+
+/**
+ * @brief This call is invoked by the peer provider to release a rx entry
+ * from owner provider's resource pool
+ *
+ * @param[in] entry the entry to be freed
+ */
+static void rxr_srx_free_entry(struct fi_peer_rx_entry *entry)
+{
+}
+
+/**
+ * @brief This call is invoked by the owner provider to start progressing
+ * the rx_entry that matches a received message.
+ *
+ * @param rx_entry the rx entry to be progressed.
+ * @return int 0 on success, a negative integer on failure.
+ */
+static int rxr_srx_start_msg(struct fi_peer_rx_entry *rx_entry)
```

Comment:
Do you think replacing `start` with `progress` better describes the function?

Reply:
I prefer to keep it as `start` to be consistent with the ops name `start_msg` in the public API.